servers/services/src/main/java/org/projectnessie/services/authz/AccessChecker.java

```diff
@@ -0,0 +1,151 @@
+
+  /**
+   * Checks whether the given role/principal is allowed to list the commit log for the given
+   * Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to list the commit log is not granted.
+   */
+  void canListCommitLog(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to commit changes against the given
+   * Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to commit changes is not granted.
+   */
+  void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to read an entity value as defined by the
+   * {@link ContentsKey} for the given Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @param key The {@link ContentsKey} to check
+   * @throws AccessControlException When the permission to read an entity value is not granted.
+   */
+  void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
+      throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to read an entity value as defined by the
+   * {@link ContentsKey} with the given contentsId for the given Branch/Tag. This check is performed
+   * after {@link AccessChecker#canReadEntityValue(AccessContext, NamedRef, ContentsKey)} while
+   * streaming {@link Contents}. The reason for having this check is that one might want to make
+   * sure that users are only allowed to see {@link Contents} at a particular point in time, which
+   * is expressed by the contentsId.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @param key The {@link ContentsKey} to check
+   * @param contentsId The ID of the {@link Contents} object
+   * @throws AccessControlException When the permission to read an entity value is not granted.
+   */
+  void canReadEntityValueWithId(
+      AccessContext context, NamedRef ref, ContentsKey key, String contentsId)
+      throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to update an entity value as defined by the
+   * {@link ContentsKey} for the given Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @param key The {@link ContentsKey} to check
+   * @throws AccessControlException When the permission to update an entity value is not granted.
+   */
+  void canUpdateEntity(AccessContext context, NamedRef ref, ContentsKey key)
```

Comment:
In my mind we should prob put contents id on these methods too. I am not certain of a situation where a user would not be able to read a `ContentsKey` (b/c of the underlying Id) but still be able to update/delete it.

Reply:
sgtm, pushed a fix